cloud/google/machineactuator_test.go

```diff
@@ -0,0 +1,119 @@
+package google_test
+
+import (
+	"github.com/stretchr/testify/assert"
+	compute "google.golang.org/api/compute/v1"
+	"sigs.k8s.io/cluster-api/cloud/google"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"strings"
+	"testing"
+)
+
+type GCEClientComputeServiceMock struct {
+	mockImagesGet           func(project string, image string) (*compute.Image, error)
+	mockImagesGetFromFamily func(project string, family string) (*compute.Image, error)
+	mockInstancesDelete     func(project string, zone string, targetInstance string) (*compute.Operation, error)
+	mockInstancesGet        func(project string, zone string, instance string) (*compute.Instance, error)
+	mockInstancesInsert     func(project string, zone string, instance *compute.Instance) (*compute.Operation, error)
+	mockZoneOperationsGet   func(project string, zone string, operation string) (*compute.Operation, error)
+}
+
+func (c *GCEClientComputeServiceMock) ImagesGet(project string, image string) (*compute.Image, error) {
+	if c.mockImagesGet == nil {
+		return nil, nil
+	}
+	return c.mockImagesGet(project, image)
+}
+
+func (c *GCEClientComputeServiceMock) ImagesGetFromFamily(project string, family string) (*compute.Image, error) {
+	if c.mockImagesGetFromFamily == nil {
+		return nil, nil
+	}
+	return c.mockImagesGetFromFamily(project, family)
+}
+
+func (c *GCEClientComputeServiceMock) InstancesDelete(project string, zone string, targetInstance string) (*compute.Operation, error) {
+	if c.mockInstancesDelete == nil {
+		return nil, nil
+	}
+	return c.mockInstancesDelete(project, zone, targetInstance)
+}
+
+func (c *GCEClientComputeServiceMock) InstancesGet(project string, zone string, instance string) (*compute.Instance, error) {
+	if c.mockInstancesGet == nil {
+		return nil, nil
+	}
+	return c.mockInstancesGet(project, zone, instance)
+}
+
+func (c *GCEClientComputeServiceMock) InstancesInsert(project string, zone string, instance *compute.Instance) (*compute.Operation, error) {
+	if c.mockInstancesInsert == nil {
+		return nil, nil
+	}
+	return c.mockInstancesInsert(project, zone, instance)
+}
+
+func (c *GCEClientComputeServiceMock) ZoneOperationsGet(project string, zone string, operation string) (*compute.Operation, error) {
+	if c.mockZoneOperationsGet == nil {
+		return nil, nil
+	}
+	return c.mockZoneOperationsGet(project, zone, operation)
+}
+
+func TestOneDisk(t *testing.T) {
```

Comment:
I may have just not seen it but is there a test for your logic which will set the min disk size if the given disk size is too small?

Reply:
Yeah, we should add this test. I will do so. Note that this is pre-existing behavior as of a few days ago where another commit effectively set the disk size to 30GB at all times.